### PR TITLE
[3.13] gh-121039: add Floats/ComplexesAreIdenticalMixin to test.support.testcase (GH-121071)

### PR DESCRIPTION
**Pull request by skirpichev**
(cherry picked from commit 8ef8354ef15e00d484ac2ded9442b789c24b11e0)

* Issue: gh-121039

### CODE REVIEW

**Review by serhiy-storchaka — APPROVED** [review]

File: Lib/test/test_ctypes/test_numbers.py [hunk 1]
```diff
@@ -39,7 +40,7 @@ def valid_ranges(*types):
 bool_values = [True, False, 0, 1, -1, 5000, 'test', [], [1]]
 
 
-class NumberTestCase(unittest.TestCase):
+class NumberTestCase(unittest.TestCase, ComplexesAreIdenticalMixin):
```
Comment (serhiy-storchaka) [inline Lib/test/test_ctypes/test_numbers.py]: Wait, is it used?

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]

**Comment by skirpichev** [comment]
I have made the requested changes; please review again.

**Review by serhiy-storchaka — APPROVED** [review]

**Review by vstinner — APPROVED** [review]
LGTM. It can wait for 3.13.1.

**Comment by skirpichev** [comment]
Depend on https://github.com/python/cpython/pull/123938

**Pull request merged**